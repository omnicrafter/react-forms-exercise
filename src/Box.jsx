const Box = ({ boxKey, color, width, height, removeBox }) => {
  return (
    <>
      <div
        style={{
          display: "inline-block",
          backgroundColor: color,
          width: `${width}px`,
          height: `${height}px`,
        }}
      ></div>
      <button onClick={() => removeBox(boxKey)}>X</button>
    </>
  );
};

export default Box;
