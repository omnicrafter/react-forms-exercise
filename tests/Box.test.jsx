import { describe, it, expect } from "vitest";
import { render, screen, fireEvent } from "@testing-library/react";
import Box from "../src/Box";

describe("Box", () => {
  it("renders without crashing", () => {
    render(<Box />);
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <Box
        boxKey="1"
        color="red"
        width="100"
        height="100"
        removeBox={() => {}}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("calls the removeBox function when the button is clicked", () => {
    const removeBox = vitest.fn();
    const { getByText } = render(
      <Box
        boxKey="1"
        color="red"
        width="100"
        height="100"
        removeBox={removeBox}
      />
    );

    fireEvent.click(getByText("X"));

    expect(removeBox).toHaveBeenCalledTimes(1);
    expect(removeBox).toHaveBeenCalledWith("1");
  });
});
