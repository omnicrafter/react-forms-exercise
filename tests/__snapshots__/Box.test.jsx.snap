// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Box > matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="display: inline-block; background-color: red; width: 100px; height: 100px;"
  />
  <button>
    X
  </button>
</DocumentFragment>
`;
